### ISSUE TITLE
Faulty Core Fix: Clash between PEAR.Functions.FunctionCallSignature.Indent and Generic.WhiteSpace.ScopeIndent.Incorrect

### ISSUE THREAD

**Issue by pento** [WordPress-Coding-Standards/WordPress-Coding-Standards#1019]
Given the following example:

```php
a( array( 'b' => 'c' ), d( array( 'e' => 'f' ) ) );
```

`phpcbf` bounces back and forth between `PEAR.Functions.FunctionCallSignature.Indent` and `Generic.WhiteSpace.ScopeIndent.Incorrect`, until it runs out of loops.

Disabling `PEAR.Functions.FunctionCallSignature.Indent` allows it to complete, but gives incorrect indentation.

The bug can be seen in Core in `tests/phpunit/tests/formatting/StripSlashesDeep.php`.

**Comment by jrfnl** [comment]
@pento Isn't that the same issue as #968 ?

**Comment by pento** [comment]
It's possibly related, but slightly different - the files affected by #968 can be made to complete a `phpcbf` run by excluding the `WordPress.Arrays.ArrayIndentation` rule. That doesn't help for this case, I also have to exclude `PEAR.Functions.FunctionCallSignature.Indent` for it to run.

I'm fine with closing this as a dupe if you think they have the same root cause, just with slightly different symptoms.

**Comment by jrfnl** [comment]
@pento You actually mentioned the sample you posted here in a comment in that thread a couple of weeks back: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/968#issuecomment-310887107 and I've investigated both the original sample of #968 as well as the second sample.
My investigation showed they have the same pattern causing the issue: associative array arguments as part of a function call and the same root cause: the existing exclusion of the two rules from `PEAR.Functions.FunctionCallSignature`.

I suggest closing this issue and would appreciate your opinion on the various directions for solving this I've previously outlined in #968.

**Comment by pento** [comment]
Hah, good catch on me leaving the same sample on #968. 🙃
